pkg/resource/table/hooks.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	"errors"
+	"time"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+var (
+	ErrTableDeleting = errors.New("table in 'DELETING' state, cannot be modified or deleted")
+	ErrTableCreating = errors.New("table in 'CREATING' state, cannot be modified or deleted")
+	ErrTableUpdating = errors.New("table in 'UPDATING' state, cannot be modified or deleted")
+)
+
+var (
+	// TerminalStatuses are the status strings that are terminal states for a
+	// DynamoDB table
+	TerminalStatuses = []v1alpha1.TableStatus_SDK{
+		v1alpha1.TableStatus_SDK_ARCHIVING,
+		v1alpha1.TableStatus_SDK_DELETING,
+	}
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		ErrTableDeleting,
+		5*time.Second,
```

Comment:
Is this an arbitrary amount? I'd like to see requeue times standardised over all controllers, perhaps?

Reply:
Not arbitrary, just remarked that tables are created super fast. The default value is 30seconds https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/requeue/requeue.go#L20-L22